src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -4975,14 +5034,299 @@ public String toString() {
         }
     }
 
-    //-------------------------------------------------------------------------
+    //-----------------------------------------------------------------------
+
     /**
-     * Length comparator.
+     * Prints or parses day periods.
      */
-    static final Comparator<String> LENGTH_SORT = new Comparator<String>() {
+    static final class DayPeriodPrinterParser implements DateTimePrinterParser {
+        private final TextStyle textStyle;
+        private final static ConcurrentMap<Locale, LocaleStore> DAYPERIOD_LOCALESTORE = new ConcurrentHashMap<>();
+
+        /**
+         * Constructor.
+         *
+         * @param textStyle  the text style, not null
+         */
+        DayPeriodPrinterParser(TextStyle textStyle) {
+            // validated by caller
+            this.textStyle = textStyle;
+        }
+
         @Override
-        public int compare(String str1, String str2) {
-            return str1.length() == str2.length() ? str1.compareTo(str2) : str1.length() - str2.length();
+        public boolean format(DateTimePrintContext context, StringBuilder buf) {
+            Long hod = context.getValue(HOUR_OF_DAY);
+            if (hod == null) {
+                return false;
+            }
+            Long moh = context.getValue(MINUTE_OF_HOUR);
+            long value = (hod * 60 + (moh != null ? moh : 0)) % 1_440;
```

Comment:
`long value = Math.floorMod(hod, 24) * 60 + (moh != null ? Math.floorMod(moh, 60) : 0);`

and remove the next three lines

Reply:
Thanks, Stephen. I made the changes based on your comments.